packages/grid/x-data-grid-pro/src/DataGridPro/useDataGridProComponent.tsx

```diff
@@ -79,6 +79,7 @@ export const useDataGridProComponent = (
   props: DataGridProProcessedProps,
 ) => {
   const apiRef = useGridInitialization(inputApiRef, props);
+  const orderedFieldsBeforePinningColumns = React.useRef<string[] | null>(null);
```

Comment:
Maybe I'm missing something but why do we need this? It's also in the state, isn't it?

Reply:
It's not part of the state, but an internal cache used by both the column pinning hook and pre-processor. I don't think it should go to the state because, when exporting/restoring the state, I don't plan to still allow to restore the pinned columns back to their original position.